hamilflow/models/pendulum.py

```diff
@@ -0,0 +1,146 @@
+import math
+from functools import cached_property
+from typing import Any, Dict, Optional, Union
+
+import numpy as np
+import pandas as pd
+from numpy.typing import ArrayLike
+from pydantic import BaseModel, Field, computed_field
+from scipy.special import ellipj, ellipk
+
+
+class PendulumSystem(BaseModel):
+    r"""The params for the pendulum
+
+    :param float omega0: $\omega_0 \coloneqq \sqrt{\frac{U}{I}} > 0$, frequency
+    parameter
+    """
+
+    omega0: float = Field(gt=0)
+
+    def __init__(self, omega0: float, **kwargs: Any) -> None:
```

Comment:
`__init__` is not required. It is in fact may create conflicts for Pydantic models.

Reply:
What I wanted is to enable positional argument, since I only have 1 argument.

Reply:
Ah I got it. In this case, I would suggest we keep the signature the same as [BaseModel](https://github.com/pydantic/pydantic/blob/a3b7214a1d6ac8e32c29e9d4436ea6a9e253ead7/pydantic/main.py#L165-L178) as use something like the following:

```suggestion
    def __init__(self, /, omega0: float, **kwargs: Any) -> None:
```


I tested some serialization of this model, it works fine.

Reply:
The slash enforces positional argument, which I also do not want. What I want is to have flexibility to use either positional or keyword arguments.

Reply:
> The slash enforces positional argument, which I also do not want. What I want is to have flexibility to use either positional or keyword arguments.

Here we only enforce self. You still have positional arguments. `*` would remove the flexibility you mentioned not `/`.

It is an antipattern to inherit from a class and break the signature consistency.

Reply:
https://github.com/kausalflow/hamilflow/pull/22/commits/bfd9a091e0a52329b7c3f4ec74d1eb6758e7d405